src/impl.ml

```diff
@@ -141,6 +141,14 @@ let create common filename size parent =
   with Failure x ->
     `Error(true, x)
 
+let open_with_creat path rw =
+  ( if not(Sys.file_exists path) then begin
```

Comment:
I don't like mixing blocking functions from `Sys` with non-blocking functions from Lwt. This should use `Lwt_unix.file_exists` instead, however that was a pre-existing problem and should be done in a separate PR.

Reply:
agreed.

Reply:
Is it wise to rely on a global variable in another library to get the desired flags? I would have liked it better to do this explicitly here, if it is so important.